src/server/common/dbconn.cpp

```diff
@@ -858,26 +861,24 @@ mariadb::statement_ref dbconn::query(const mariadb::connection_ref& connection,
 	unsigned i = 0;
 	for(const auto& param : params)
 	{
-		if(std::holds_alternative<bool>(param))
-		{
-			stmt->set_boolean(i, std::get<bool>(param));
-		}
-		else if(std::holds_alternative<int>(param))
-		{
-			stmt->set_signed32(i, std::get<int>(param));
-		}
-		else if(std::holds_alternative<unsigned long long>(param))
-		{
-			stmt->set_signed64(i, std::get<unsigned long long>(param));
-		}
-		else if(std::holds_alternative<std::string>(param))
-		{
-			stmt->set_string(i, std::get<std::string>(param));
-		}
-		else if(std::holds_alternative<const char*>(param))
-		{
-			stmt->set_string(i, std::get<const char*>(param));
-		}
+		std::visit([&](const auto& p) {
```

Comment:
I'd rather be explicit about what is captured.

```suggestion
		std::visit([&i, &stmt](const auto& p) {
```